example/Main.hs

```diff
@@ -183,15 +183,16 @@ reactor lf inp = do
                                  . J.uri
             fileName =  J.uriToFilePath doc
         liftIO $ U.logs $ "********* fileName=" ++ show fileName
-        sendDiagnostics doc (Just 0)
+        sendDiagnostics (J.toNormalizedUri doc) (Just 0)
 
       -- -------------------------------
 
       HandlerRequest (NotDidChangeTextDocument notification) -> do
-        let doc :: J.Uri
+        let doc :: J.NormalizedUri
             doc  = notification ^. J.params
                                  . J.textDocument
                                  . J.uri
+                                 . to J.toNormalizedUri
```

Comment:
For now, are we just normalising didchangetextdocument notifications?

Reply:
We are normalizing everything in the VFS and everything in the DiagnosticsStore.

Reply:
This sounds good to me. How breaking would it be to make this the default behaviour for `URL`?
Could we also define a custom `Eq` instance rather than have a smart constructor to preserve roundtripping?